pkgs/applications/misc/bb/default.nix

```diff
@@ -9,19 +9,24 @@ stdenv.mkDerivation rec {
     sha256 = "1i411glxh7g4pfg4gw826lpwngi89yrbmxac8jmnsfvrfb48hgbr";
   };
 
+  patches = [
+    # add / update include files to get function prototypes
+    ./included-files-updates.diff
+  ];
+
+  nativeBuildInputs = [ autoreconfHook ];
```

Comment:
Is this necessary on darwin? The original configure script seems ok to me, at least on GNU.

Reply:
yes, without it the configure script fails
```
checking whether the C compiler (clang  ) works... no
configure: error: installation or configuration problem: C compiler cannot create executables.
```
https://hydra.nixos.org/build/273786787/log/tail

the old configure creates a `main` function with a return type or something which causes clang to error out, but when the configure script is regenerated it is fixed.

note that this is also needed for gcc14 as
```
nb -E 'with import ./.{}; bb.override { stdenv = gcc14Stdenv; }
```
will break on master without `autoreconfHook`.

Reply:
and gcc14 needs the header modifications to get the function prototypes as well otherwise it error out like clang does. so without this change bb will break on linux once gcc is bumped to gcc14 sometime next year. I know they tried to bump it a month ago but ran into a gcc bug and had to revert.

Reply:
Ok, I guess it's time to retire the 26 years old configure script.